internal/commands/config_registries.go

```diff
@@ -0,0 +1,173 @@
+package commands
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+
+	"github.com/buildpacks/pack/internal/config"
+	"github.com/buildpacks/pack/internal/slices"
+	"github.com/buildpacks/pack/internal/stringset"
+	"github.com/buildpacks/pack/internal/style"
+	"github.com/buildpacks/pack/logging"
+	"github.com/buildpacks/pack/registry"
+)
+
+var (
+	bpRegistryExplanation = "A Buildpack Registry is a (still experimental) place to publish, store, and discover buildpacks. "
+)
+
+var (
+	setDefault   bool
+	registryType string
+)
+
+func ConfigRegistries(logger logging.Logger, cfg config.Config, cfgPath string) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "registries",
+		Aliases: []string{"registry", "registreis"},
+		Short:   prependExperimental("Interact with registries"),
+		RunE: logError(logger, func(cmd *cobra.Command, args []string) error {
+			listRegistries(args, logger, cfg)
+			return nil
+		}),
+	}
+
+	addCmd := generateAdd("registries", logger, cfg, cfgPath, addRegistry)
+	addCmd.Args = cobra.ExactArgs(2)
+	addCmd.Example = "pack config registries add my-registry https://github.com/buildpacks/my-registry"
+	addCmd.Short = prependExperimental(addCmd.Short)
+	addCmd.Long = bpRegistryExplanation + "Users can add registries from the config by using registries remove, and publish/yank buildpacks from it, as well as use those buildpacks when building applications."
+	addCmd.Flags().BoolVar(&setDefault, "default", false, "Set this buildpack registry as the default")
+	addCmd.Flags().StringVar(&registryType, "type", "github", "Type of buildpack registry [git|github]")
+	cmd.AddCommand(addCmd)
```

Comment:
With the number of changes I have to do to the generated command, I'm not sure whether it would be better to just define the command anew, instead of generating it. Thoughts?